libevmasm/Assembly.cpp

```diff
@@ -756,3 +759,31 @@ LinkerObject const& Assembly::assemble() const
 	}
 	return ret;
 }
+
+vector<size_t> Assembly::decodeSubIds(u256 _data)
+{
+	size_t length = static_cast<size_t>((_data >> 248) & 0xFF);
+	if (length == 0)
+		length = 1;
+	std::vector<size_t> subIds;
+	u256 const mask = 0xFF;
+	while (length--)
+	{
+		auto currentSubId = static_cast<size_t>(_data & mask);
+		subIds.emplace_back(currentSubId);
+		_data >>= 8;
+	}
+	return subIds;
+}
+
+u256 Assembly::encodeSubIds(std::vector<size_t> const& _subIds)
+{
+	u256 encodedSubId = u256(_subIds.size()) << 248;
+	for (size_t subIndex = 0; subIndex < _subIds.size(); ++subIndex)
+	{
+		u256 subId = _subIds[subIndex];
+		assertThrow(subId <= 512, AssemblyException, "");
```

Comment:
Why 512 when it is encoded as 8-bits?

Reply:
Mistake, wasn't thinking too much. I think I had an idea to use 9 bits...